optimum/onnx/configuration.py

```diff
@@ -0,0 +1,256 @@
+        """
+        common_outputs = self._onnx_config.outputs
+        common_outputs.update(self._tasks_to_extra_outputs["default"])
+        if "loss" in common_outputs.keys():
+            common_outputs.move_to_end("loss", last=False)
+        return copy.deepcopy(common_outputs)
+
+    def generate_dummy_inputs(
+        self,
+        preprocessor: Union["PreTrainedTokenizerBase", "FeatureExtractionMixin"],
+        batch_size: int = -1,
+        seq_length: int = -1,
+        is_pair: bool = False,
+        framework: Optional[TensorType] = None,
+        num_channels: int = 3,
+        image_width: int = 40,
+        image_height: int = 40,
+        tokenizer: "PreTrainedTokenizerBase" = None,
+    ) -> Mapping[str, Any]:
+        """
+        Generate inputs to provide to the ONNX exporter for the specific framework
+        Args:
+            preprocessor: ([`PreTrainedTokenizerBase`] or [`FeatureExtractionMixin`]):
+                The preprocessor associated with this model configuration.
+            batch_size (`int`, *optional*, defaults to -1):
+                The batch size to export the model for (-1 means dynamic axis).
+            seq_length (`int`, *optional*, defaults to -1):
+                The sequence length to export the model for (-1 means dynamic axis).
+            is_pair (`bool`, *optional*, defaults to `False`):
+                Indicate if the input is a pair (sentence 1, sentence 2)
+            framework (`TensorType`, *optional*, defaults to `None`):
+                The framework (PyTorch or TensorFlow) that the tokenizer will generate tensors for.
+            num_channels (`int`, *optional*, defaults to 3):
+                The number of channels of the generated images.
+            image_width (`int`, *optional*, defaults to 40):
+                The width of the generated images.
+            image_height (`int`, *optional*, defaults to 40):
+                The height of the generated images.
+        Returns:
+            Mapping[str, Tensor] holding the kwargs to provide to the model's forward function
+        """
+        # Generate dummy labels
+        dummy_inputs = super().generate_dummy_inputs(
+            preprocessor,
+            batch_size,
+            seq_length,
+            is_pair,
+            framework,
+            num_channels,
+            image_width,
+            image_height,
+            tokenizer,
+        )
+        for label, input in self._tasks_to_extra_inputs[self.task].items():
+            batch_size = compute_effective_axis_dimension(
+                batch_size, fixed_dimension=self.default_fixed_batch, num_token_to_add=0
+            )
+            if "sequence" in input.values():
+                seq_length = compute_effective_axis_dimension(
+                    seq_length, fixed_dimension=self.default_fixed_sequence, num_token_to_add=0
+                )
+                if framework == TensorType.PYTORCH:
+                    import torch
```

Comment:
In transformers, we check that the framework is available.

```suggestion
                    if not torch_is_available():
                        raise ValueError("Could not generate dummy inputs to perform the ONNX export through PyTorch because not installation was found.")
                    else:
                        import torch
```

We have a similar `is_tf_available` method for TensorFlow.